### PR TITLE
[ERP-881] Instruct browsers to not use automatic translations

### PR DESCRIPTION
**Pull request by Lexiel46**

### CODE REVIEW

File: rdrf/rdrf/templates/rdrf_cdes/navbar_links.html [hunk 1]
```diff
@@ -6,13 +6,45 @@
 {% load has_subscribable_email %}
 
 {% include "localisation/language_picker_form.html" %}
+
+<script type="text/javascript">
```
Comment (ScriptSmith) [inline rdrf/rdrf/templates/rdrf_cdes/navbar_links.html]: Might be a miscommunication of requirements here, "feature toggle" was referring to a registry feature. I like your take on it though, although it appears to be a bit broken for me on Firefox (the value gets stuck)

**Review by ppettitau — COMMENTED** [review]
I'll come back and review once the "registry feature" based feature toggle is in place.

File: rdrf/rdrf/templatetags/no_translation.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+@register.filter
+def no_translation(registry_code, request):
+    def get_metadata_features(registry_model):
+        return registry_model.metadata.get('features', [])
+
+    all_registries = Registry.objects.all()
+    user = request.user
+    if registry_code in ['', None]:
```
Comment (ppettitau) [inline rdrf/rdrf/templatetags/no_translation.py]: Could also say `if not registry_code`, as empty strings and Nones are both considered falsy. This is fine though.

> The following are falsy values in Python:
> 
> The number zero (0)
> An empty string ''
> False
> None
> An empty list []
> An empty tuple ()
> An empty dictionary {}
https://www.pythontutorial.net/python-basics/python-boolean/

File: rdrf/rdrf/templatetags/no_translation.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from rdrf.models.definition.models import Registry
+from registry.groups.models import CustomUser
+from django.contrib.auth.models import AnonymousUser
+
+register = template.Library()
+
+
+@register.filter
+def no_translation(registry_code, request):
+    def get_metadata_features(registry_model):
```
Comment (ppettitau) [inline rdrf/rdrf/templatetags/no_translation.py]: Can use `registry_model.features` instead.

File: rdrf/rdrf/templatetags/no_translation.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        if all_registries.count() == 1:
+            return 'no_translation' in get_metadata_features(all_registries.first())
+        if not isinstance(user, AnonymousUser):
+            for registry in CustomUser.objects.filter(username=user).values('registry'):
+                if 'no_translation' not in get_metadata_features(all_registries.filter(id=registry['registry'])[0]):
+                    return False
+            return True
+        return False
+    else:
+        return 'no_translation' in get_metadata_features(all_registries.filter(code=registry_code)[0])
```
Comment (ppettitau) [inline rdrf/rdrf/templatetags/no_translation.py]: This is a bit hard to follow. Can we clean this up so it's clearer?
Some general feedback about this function:
* If the function returns a boolean, it is convention to prefix it with a is/has for readability
* We could make this more generic by passing in a feature. This allows it to be reused for other purposes later if required. It also help reduce duplication.
* When we have a boolean condition, it is generally more readable to have the 'truthy' path first. This increases readability and helps to avoid double negatives etc
* If the user is the only thing needed from the `request` object, then just pass in the user to the function
* Depending on the downstream affects to the templates from changing the signature of this tag, it might be worth having a second tag specifically for `translate` . Perhaps you could have a tag that returns the translate attributes (empty for default behaviour, "no" for registries opting for no translations).

e.g.  (FYI this is untested)
```
def has_feature(registry_code, user, feature):
    all_registries = Registry.objects.all()
    registry = None
    
    if registry_code:
    	registry = all_registries.get(code=registry_code)
    elif all_registries.count() == 1:
        registry = all_registries.first()
    elif not user.is_anonymous:
	registry = user.my_registry

    return registry and feature in registry.features
```

**Review by ppettitau — CHANGES_REQUESTED** [review]

File: rdrf/rdrf/templatetags/is_no_translation.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from django import template
+from rdrf.models.definition.models import Registry
+
+register = template.Library()
+
+
+@register.filter
+def is_no_translation(registry):
```
Comment (ppettitau) [inline rdrf/rdrf/templatetags/is_no_translation.py]: No longer a boolean return type so update the method name e.g. `get_translate_value`

**Review by ppettitau — CHANGES_REQUESTED** [review]

File: rdrf/rdrf/templatetags/get_translate_value.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+from django import template
+from rdrf.models.definition.models import Registry
+
+register = template.Library()
+
+
+@register.filter
+def get_translate_value(registry):
+    if registry:
+        return 'no' if registry.has_feature('no_translation') else 'yes'
```
Comment (ScriptSmith) [inline rdrf/rdrf/templatetags/get_translate_value.py]: Should be added to:

https://github.com/eresearchqut/trrf/blob/9f51c60e4c3cd27d27665c513d440882b1b3dee9/rdrf/rdrf/helpers/registry_features.py#L1-L22

**Review by ScriptSmith — CHANGES_REQUESTED** [review]

**Review by ScriptSmith — APPROVED** [review]

File: rdrf/rdrf/templatetags/get_translate_value.py [hunk 7]
```diff
@@ -1,12 +1,14 @@
 from django import template
 from rdrf.models.definition.models import Registry
+from rdrf.helpers.registry_features import RegistryFeatures
 
 register = template.Library()
 
 
 @register.filter
 def get_translate_value(registry):
+    no_translation = RegistryFeatures.NO_TRANSLATION
```
Comment (ppettitau) [inline rdrf/rdrf/templatetags/get_translate_value.py]: nit: You could just inline this (i.e. use `RegistryFeatures.NO_TRANSLATION` instead of `no_translation`) rather than having a variable. I don't think much is gained by having this in a variable.

**Review by ppettitau — APPROVED** [review]
Just a little nitpick feedback, but either way it looks good :+1:
